olive/model/__init__.py

```diff
@@ -93,41 +98,28 @@ def set_resource(self, resource_name: str, resource_path: Union[Path, str, Resou
             raise ValueError(f"{resource_name} is not a valid resource name.")
         if self.resource_paths[resource_name]:
             logger.debug(f"Overriding {resource_name} from {self.resource_paths[resource_name]} to {resource_path}.")
-        self.resource_paths[resource_name] = create_resource_path(resource_path)
-        self.local_resource_paths[resource_name] = None
 
-    def get_local_resource(self, resource_name: str) -> str:
+        if resource_path is not None:
+            resolved_resource_path = create_resource_path(resource_path)
+            assert (
+                resolved_resource_path.is_local_resource_or_string_name()
+            ), f"{resource_name} must be local path or string name."
+            resource_path = resolved_resource_path.get_path()
+
+        self.resource_paths[resource_name] = resource_path
+
+    def get_resource(self, resource_name: str) -> str:
         """
         Get local path of a resource.
 
         :param resource_name: name of the resource.
         :return: local path.
```

Comment:
The `return` may be not be local path, right?

Reply:
the resource always be local path

Reply:
Ic, looking other parts now.